client/my-sites/upgrades/cart/cart-coupon.jsx

```diff
@@ -77,22 +77,26 @@ module.exports = React.createClass({
 			return;
 		}
 
+		if ( this.state.hasSubmittedCoupon ) {
+			return;
+		}
+
 		return (
-			<form onSubmit={ this.applyCoupon }>
+			<div className="form">
 				<input type="text" placeholder={ this.translate( 'Enter Coupon Code', { textOnly: true } ) } onChange={ this.handleCouponInput } value={ this.state.couponInputValue } />
-				<button type="submit" className="button">
+				<button type="button" className="button" onClick={ this.applyCoupon }>
```

Comment:
Actually, no, [it's `submit`](https://developer.mozilla.org/en/docs/Web/HTML/Element/button) :) I had to change it to `button`, because otherwise it'd submit the form it was placed in (e.g. `CreditCardPaymentBox`).